crates/bitwarden/src/vault/cipher/login.rs

```diff
@@ -101,21 +101,152 @@ pub struct Fido2Credential {
 #[serde(rename_all = "camelCase", deny_unknown_fields)]
 #[cfg_attr(feature = "mobile", derive(uniffi::Record))]
 pub struct Fido2CredentialView {
-    pub credential_id: SensitiveString,
-    pub key_type: SensitiveString,
-    pub key_algorithm: SensitiveString,
-    pub key_curve: SensitiveString,
-    pub key_value: SensitiveString,
-    pub rp_id: SensitiveString,
-    pub user_handle: Option<SensitiveString>,
-    pub user_name: Option<SensitiveString>,
-    pub counter: SensitiveString,
-    pub rp_name: Option<SensitiveString>,
-    pub user_display_name: Option<SensitiveString>,
-    pub discoverable: SensitiveString,
+    pub credential_id: DecryptedString,
+    pub key_type: DecryptedString,
+    pub key_algorithm: DecryptedString,
+    pub key_curve: DecryptedString,
+    // This value doesn't need to be returned to the client
+    // so we keep it encrypted until we need it
+    pub key_value: EncString,
+    pub rp_id: DecryptedString,
+    pub user_handle: Option<DecryptedVec>,
+    pub user_name: Option<DecryptedString>,
+    pub counter: DecryptedString,
+    pub rp_name: Option<DecryptedString>,
+    pub user_display_name: Option<DecryptedString>,
+    pub discoverable: DecryptedString,
+    pub creation_date: DateTime<Utc>,
+}
+
+// This is mostly a copy of the Fido2CredentialView, but with the key exposed
+// Only meant to be used internally and not exposed to the outside world
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub(crate) struct Fido2CredentialFullView {
```

Comment:
My thinking here is this type is never to be exposed to the clients, so it would be strange if the clients had `Fido2Credential` <-> `Fido2CredentialLimitedView`, but no `Fido2CredentialView`

I don't like the name very much either, and I might try to delete this type and replace it by `(Fido2CredentialView, SecretKey)` anyway.